lib/ansible/module_utils/network/iosxr/iosxr.py

```diff
@@ -428,20 +436,36 @@ def run_command(module, commands):
     conn = get_connection(module)
     responses = list()
     for cmd in to_list(commands):
+
         try:
-            cmd = json.loads(cmd)
-            command = cmd['command']
-            prompt = cmd['prompt']
-            answer = cmd['answer']
+            if isinstance(cmd, str):
+                cmd = json.loads(cmd)
+            command = cmd.get('command', None)
+            prompt = cmd.get('prompt', None)
+            answer = cmd.get('answer', None)
+            sendonly = cmd.get('sendonly', False)
+            newline = cmd.get('newline', True)
         except:
             command = cmd
             prompt = None
             answer = None
+            sendonly = False
+            newline = True
 
-        out = conn.get(command, prompt, answer)
+        out = conn.get(command, prompt=prompt, answer=answer, sendonly=sendonly, newline=newline)
 
         try:
             responses.append(to_text(out, errors='surrogate_or_strict'))
         except UnicodeError:
             module.fail_json(msg=u'failed to decode output from {0}:{1}'.format(cmd, to_text(out)))
     return responses
+
+
+def put_file(module, src, dst, proto='scp'):
+    conn = get_connection(module)
+    conn.put_file_cliconf(source=src, destination=dst, proto=proto)
```

Comment:
Can we rename this to `copy_file` and `get_file`. Can't think of a reason why API name should have connection type in it.